packages/request-client.js/src/api/payment-network/erc20/proxy-info-retriever.ts

```diff
@@ -0,0 +1,90 @@
+import { ethers } from 'ethers';
+import * as Types from '../../../types';
+
+// The ERC20 proxy smart contract ABI fragment containing TransferWithReference event
+const erc20proxyContractAbiFragment = [
+  'event TransferWithReference(address tokenAddress,address to,uint256 amount,bytes indexed paymentReference)',
+];
+
+/**
+ * Retrieves a list of transfer events for an address
+ */
+export default class ProxyERC20InfoRetriever
+  implements Types.IPaymentNetworkInfoRetriever<Types.ERC20PaymentNetworkEvent> {
+  public contractProxy: ethers.Contract;
+  public provider: ethers.providers.Provider;
+
+  /**
+   * @param tokenContractAddress The address of the ERC20 contract
+   * @param address Address of the balance we want to check
+   * @param eventName Indicate if it is an address for payment or refund
+   * @param network The Ethereum network to use
+   */
+  constructor(
+    private paymentReference: string,
```

Comment:
What are those private keywords ?

Reply:
declare and assign directly the attributes to the class